javalib/src/main/scala/java/util/Base64.scala

```diff
@@ -0,0 +1,486 @@
+
+  private val mimeEncoder = new Encoder(basicEncodeTable, mimeLineLength, mimeLineSeparators)
+  private val mimeDecoder = new Decoder(basicDecodeTable, ignoreInvalid = true)
+
+  private val urlSafeEncoder = new Encoder(urlSafeEncodeTable)
+  private val urlSafeDecoder = new Decoder(urlSafeDecodeTable, ignoreInvalid = false)
+
+  // --------------------------------------------------------------------------
+
+  def getEncoder(): Encoder = basicEncoder
+
+  def getUrlEncoder(): Encoder = urlSafeEncoder
+
+  def getMimeEncoder(): Encoder = mimeEncoder
+
+  def getMimeEncoder(lineLength: Int, lineSeparator: Array[Byte]): Encoder = {
+    lineSeparator.foreach { b =>
+      if (basicDecodeTable(b & 0xff) != -1)
+        throw new IllegalArgumentException("Illegal base64 line separator character 0x" + Integer.toString(b, 16))
+    }
+    new Encoder(basicEncodeTable, lineLength / 4 * 4, lineSeparator)
+  }
+
+  def getDecoder(): Decoder = basicDecoder
+
+  def getUrlDecoder(): Decoder = urlSafeDecoder
+
+  def getMimeDecoder(): Decoder = mimeDecoder
+
+  // --------------------------------------------------------------------------
+
+  class Decoder private[Base64](table: Array[Int], ignoreInvalid: Boolean) {
+
+    def decode(src: Array[Byte]): Array[Byte] = {
+      val dst = new Array[Byte](dstRequiredLength(src, 0, src.length))
+      doDecode(new Wrapper(src), new Wrapper(dst))
+      dst
+    }
+
+    def decode(src: String): Array[Byte] = decode(src.getBytes(StandardCharsets.ISO_8859_1))
+
+    def decode(src: Array[Byte], dst: Array[Byte]): Int = {
+      if (dst.length >= dstMaxLength(src.length)) // dst is definitely large enough
+        doDecode(new Wrapper(src), new Wrapper(dst))
+      else if (dst.length < dstRequiredLength(src, 0, src.length))
+        throw new IllegalArgumentException("Output byte array is too small for decoding all input bytes")
+      else
+        doDecode(new Wrapper(src), new Wrapper(dst))
+    }
+
+    def decode(buffer: ByteBuffer): ByteBuffer = {
+      val start = buffer.position()
+      try {
+        val src = buffer.array()
+        val offset = buffer.arrayOffset() + buffer.position()
+        val length = buffer.arrayOffset() + buffer.limit()
+        val dst = new Array[Byte](dstRequiredLength(src, offset, length))
+        val written = doDecode(new Wrapper(src, offset, length), new Wrapper(dst))
+        ByteBuffer.wrap(dst, 0, written)
+      } catch {
+        case e: IllegalArgumentException =>
+          buffer.position(start)
+          throw e
+      }
+    }
+
+    def wrap(is: InputStream): InputStream = new DecodingInputStream(is, table, ignoreInvalid)
+
+    // ------------------------------------------------------------------------
+    // PRIVATE
+    // ------------------------------------------------------------------------
+
+    private def doDecode(src: Wrapper[Byte], dst: Wrapper[Byte]): Int = {
+      val srcBuffer = new Wrapper[Int](new Array[Int](4))
```

Comment:
This is the only `Wrapper[Int]` that you have. All the other ones are `Wrapper[Byte]`. You should also make this one a `Wrapper[Byte]`, and then you can actually hard-code `Wrapper` for `Byte`, eliminating the need for `@specialized`.